pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/ConsistentDataPushUtils.java

```diff
@@ -0,0 +1,296 @@
+      Map<URI, List<String>> uriToSegmentsFrom, List<String> segmentsTo)
+      throws Exception {
+    Map<URI, String> uriToLineageEntryIdMap = new HashMap<>();
+    String rawTableName = spec.getTableSpec().getTableName();
+    Map<URI, URI> segmentsUris = getStartReplaceSegmentUris(spec, rawTableName);
+    AuthProvider authProvider = AuthProviderUtils.makeAuthProvider(spec.getAuthToken());
+    LOGGER.info("Start replace segment URIs: " + segmentsUris);
+
+    int attempts = 1;
+    long retryWaitMs = 1000L;
+
+    for (Map.Entry<URI, URI> entry : segmentsUris.entrySet()) {
+      URI controllerUri = entry.getKey();
+      URI startSegmentUri = entry.getValue();
+      List<String> segmentsFrom = uriToSegmentsFrom.get(controllerUri);
+
+      if (!Collections.disjoint(segmentsFrom, segmentsTo)) {
+        // This scenario could happen only when "disable.unique.segment.name" is enabled,
+        // i.e. newly generated segment name is the same as the existing one in the table. If so, we should
+        //    1) either clean up all the consistent push custom config from table config,
+        //    2) or remove "disable.unique.segment.name" to continue enabling consistent push.
+        String errorMsg =
+            String.format("Found same segment names when attempting to enable consistent push for table: %s",
+                rawTableName);
+        LOGGER.error(errorMsg);
+        throw new RuntimeException(errorMsg);
+      }
+
+      StartReplaceSegmentsRequest startReplaceSegmentsRequest =
+          new StartReplaceSegmentsRequest(segmentsFrom, segmentsTo);
+      RetryPolicies.exponentialBackoffRetryPolicy(attempts, retryWaitMs, 5).attempt(() -> {
+        try {
+          SimpleHttpResponse response =
+              FILE_UPLOAD_DOWNLOAD_CLIENT.startReplaceSegments(startSegmentUri, startReplaceSegmentsRequest,
+                  authProvider);
+
+          String responseString = response.getResponse();
+          LOGGER.info(
+              "Got response {}: {} while sending start replace segment request for table: {}, uploadURI: {}, request:"
+                  + " {}", response.getStatusCode(), responseString, rawTableName, startSegmentUri,
+              startReplaceSegmentsRequest);
+          String segmentLineageEntryId =
+              JsonUtils.stringToJsonNode(responseString).get("segmentLineageEntryId").asText();
+          uriToLineageEntryIdMap.put(controllerUri, segmentLineageEntryId);
+          return true;
+        } catch (SocketTimeoutException se) {
+          // In case of the timeout, we should re-try.
+          return false;
+        } catch (HttpErrorStatusException e) {
+          if (e.getStatusCode() >= 500) {
+            return false;
+          } else {
+            if (e.getStatusCode() == Response.Status.NOT_FOUND.getStatusCode()) {
+              LOGGER.error("Table: {} not found when sending request: {}", rawTableName, startSegmentUri);
```

Comment:
What don't we log all status code? If things break for some other reason we want to make it easier to ops to figure what went wrong